supported-os-installer.sh

```diff
@@ -0,0 +1,524 @@
+		sudo DEBIAN_FRONTEND=noninteractive apt-get upgrade -qq -y < /dev/null > /dev/null
+	fi
+
+	list_essentials_ubuntu=(
+		apt-utils netcat-openbsd sudo libidn11 libpcre3 libgmp10 \
+		libexpat1 libstdc++6 libaio1 resolvconf \
+		unzip pax sysstat sqlite3 dnsmasq lsb-release net-tools \
+		netfilter-persistent dnsutils iptables sed wget rsyslog ldapscripts
+	)
+	list_essentials_redhat=(
+		netcat-openbsd sudo libidn11 libpcre3 libgmp10 \
+		libexpat1 libstdc++6 libaio1 resolvconf \
+		unzip pax sysstat sqlite3 dnsmasq lsb-release net-tools \
+		netfilter-persistent dnsutils iptables sed wget rsyslog ldapscripts
+	)
+
+	# Prints all package names in a single line to trigger install.
+	all_pkgs_ubuntu="$(echo ${list_essentials_ubuntu[*]})"
+	all_pkgs_redhat="$(echo ${list_essentials_redhat[*]})"
+	if [[ "$redhat_release" == "true" ]]; then
+		sudo yum install -q -y $all_pkgs_redhat< /dev/null > /dev/null
+	else
+		sudo DEBIAN_FRONTEND=noninteractive apt-get install -qq -y $all_pkgs_ubuntu< /dev/null > /dev/null
+	fi
+	[[ "$?" -ne "0" ]] && echo "Cannot get some of essential packages. Installation may fail." || \
+	echo "Essential packages installed successfully."
+}
+
+function install_dns {
+	# Install a DNS Server
+	echo "Configuring dnsmasq ..."
+	sudo mv /etc/dnsmasq.conf /etc/dnsmasq.conf.old || true
+
+	#create the conf file
+	nsentry="$(printf "server=8.8.8.8\nserver=8.8.4.4\nserver=9.9.9.9\nserver=149.112.112.112\nserver=1.1.1.1\nserver=1.0.0.1\nlisten-address=127.0.0.1\ndomain='$DOMAIN_NAME'\nmx-host='$DOMAIN_NAME','$HOSTNAME',0\naddress=/$HOSTNAME/$HOST_IP\n")"
+	echo "$nsentry" | sudo tee /etc/dnsmasq.conf
+	local resolv_file="/etc/resolv.conf"
+	grep 'nameserver 8.8.8.8' /etc/resolv.conf >> /dev/null
+	local ret="$(echo $?)"
+	if [[ "$ret" -eq 0 ]]; then
+		echo "Nameserver is correct in $resolv_file"
+	else
+		# Edit resolve.conf
+		sudo cp /etc/resolv.conf /etc/resolv.conf.old
+		sudo sed -i 's/^nameserver [0-9]\+\.[0-9]\+\.[0-9]\+\.[0-9]\+/nameserver 8.8.8.8/' "$resolv_file"
+		echo "Replacement completed!"
+	fi
+
+	# restart dns services
+	sudo systemctl enable dnsmasq.service > /dev/null 2>&1 \
+	&& sudo systemctl restart dnsmasq.service
+	if [[ "$?" -eq 0 ]]; then
+		echo "dnsmasq.service started successfully."
+	else
+		echo "Failed to start service -> dnsmasq.service" && exit 1;
+	fi
+
+	# Check DNS
+	echo "Checking DNS ..."
+	name="$(host license.zimbra.com)"
+	if [[ "$name" == *"not found"* ]]; then
+		echo -e "DNS resolution failed! Check your resolve.conf file." && exit 1
+	else
+		echo -e "DNS resolution done."
+	fi
+}
+
+function enable_ports {
+	# Enable all required ports using iptables.
+	echo "Enabling firewall for Zimbra ports ..."
+	echo "Ports 22 / 25 / 143 / 80 / 443 / 465 / 587 / 995 / 993 / 9071 will be opened to the internet."
+	echo "Please check your iptables for more info."
+
+	# Array of all ports required.
+	ports_arr=("25" "80" "110" "143" "443" "465" "587" "993" "995" "7071" "9071")
```

Comment:
not sure if we should open 7071 port

Reply:
Yes, 9071 is already being opened!
Not opening the port 7071 in the script!